### PR TITLE
kd: add daemon subcommand

### PR DESCRIPTION
**Pull request by rjeczalik**

### CODE REVIEW

File: go/src/koding/klientctl/daemon/daemon.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return errors.New(url + ":" + http.StatusText(resp.StatusCode))
```
Comment (ppknap) [inline go/src/koding/klientctl/daemon/daemon.go]: Please do either `f.Close()` or `defer f.Close()` after file is opened.

File: go/src/koding/klientctl/main.go [hunk 2]
```diff
@@ -329,6 +338,79 @@ func run(args []string) {
...
 		}, {
+			Name:  "daemon",
+			Usage: "Manage KD Daemon service.",
+			Subcommands: []cli.Command{{
+				Name:   "install",
+				Usage:  "Install the daemon and dependencies.",
+				Action: ctlcli.ExitErrAction(DaemonInstall, log, "install"),
+				Flags: []cli.Flag{
+					cli.BoolFlag{
+						Name:  "force, y",
```
Comment (ppknap) [inline go/src/koding/klientctl/main.go]: What about `--force-yes` like in Debian's `apt-get`pm?

File: go/src/koding/klientctl/main.go [hunk 3]
```diff
@@ -540,6 +559,12 @@ func run(args []string) {
 		find(app.Commands, "machine", "umount"),
 		find(app.Commands, "machine", "exec"),
 		find(app.Commands, "machine", "cp"),
+		find(app.Commands, "daemon", "install"),
+		find(app.Commands, "daemon", "uninstall"),
+		find(app.Commands, "daemon", "update"),
+		find(app.Commands, "daemon", "start"),
```
Comment (ppknap) [inline go/src/koding/klientctl/main.go]: just a note: these will be confusing after we add `kd machine stop/start`.

File: go/src/koding/klientctl/daemon/client.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+// Starts starts KD daemon.
+func (c *Client) Start() error {
+	c.init()
+
+	svc, err := c.d.service()
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Starting daemon service ... ")
```
Comment (rjeczalik) [inline go/src/koding/klientctl/daemon/client.go]: Done.

File: go/src/koding/klientctl/daemon/client.go [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+
+// Ping probes KD daemon and returns nil error when it's ready.
+func (c *Client) Ping() error {
+	timeout := time.NewTimer(c.timeout())
+	defer timeout.Stop()
+
+	tick := time.NewTicker(time.Second)
+	defer tick.Stop()
+
+	var err error
```
Comment (rjeczalik) [inline go/src/koding/klientctl/daemon/client.go]: Done.

File: go/src/koding/klientctl/daemon/client.go [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+func (c *Client) readCache() {
+	c.d = newDetails()
+
+	// Ignoring read error, if it's non-nil then empty cache is going to
+	// be used instead.
+	_ = c.store().Commit(func(cache *config.Cache) error {
+		return cache.GetValue("daemon.details", c.d)
+	})
+
+	if c == DefaultClient {
```
Comment (rjeczalik) [inline go/src/koding/klientctl/daemon/client.go]: Done.

File: go/src/koding/klientctl/daemon/daemon.go [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+	klientHome := "/opt/kite/klient"
+	if s := os.Getenv("KLIENT_HOME"); s != "" {
+		klientHome = s
+	}
+
+	return &Details{
+		Username:   config.CurrentUser.Username,
+		KodingHome: config.KodingHome(),
+		KlientHome: klientHome,
+		Osxfuse: &Package{
```
Comment (rjeczalik) [inline go/src/koding/klientctl/daemon/daemon.go]: Simplicity, no need for a map now.

File: go/src/koding/klientctl/daemon/daemon.go [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+	if !d.Base.IsNil() {
+		return d.Base.URL
+	}
+	return conf.Konfig.Endpoints.Koding.Public.URL
+}
+
+func cmd(name string, args ...string) *exec.Cmd {
+	c := exec.Command(name, args...)
+	c.Stderr = os.Stderr
+	c.Stdout = os.Stderr
```
Comment (rjeczalik) [inline go/src/koding/klientctl/daemon/daemon.go]: Yes.

File: go/src/koding/klientctl/daemon/daemon.go (hunk 1 above)
Comment (rjeczalik) [inline go/src/koding/klientctl/daemon/daemon.go]: Done.

File: go/src/koding/klientctl/main.go (hunk 2 above)
Comment (rjeczalik) [inline go/src/koding/klientctl/main.go]: Done.

**Review by ppknap — APPROVED** [review]

**Comment by rjeczalik** [comment]
deps: https://github.com/koding/koding/pull/10962

**Pull request merged**